diff --git a/satip.c b/satip.c
index ec2f8ab..4b6fb4e 100644
--- a/satip.c
+++ b/satip.c
@@ -5,6 +5,7 @@
  *
  */
 
+#include <ctype.h>
 #include <getopt.h>
 #include <vdr/plugin.h>
 #include "common.h"
diff --git a/socket.c b/socket.c
index 4cb19f3..011129e 100644
--- a/socket.c
+++ b/socket.c
@@ -289,7 +289,15 @@ int cSatipSocket::ReadMulti(unsigned char *bufferAddrP, unsigned int *elementRec
      error("%s Invalid parameter(s)", __PRETTY_FUNCTION__);
      return -1;
      }
+#if defined(__GLIBC__) // musl-libc evaluates the whole line at once
 #if defined(__GLIBC_PREREQ) && __GLIBC_PREREQ(2,12)
+#define ENABLE_RECVMMSG
+#endif
+#else
+#define ENABLE_RECVMMSG // use musl recvmmsg()
+#endif
+
+#if defined(ENABLE_RECVMMSG)
   // Initialize iov and msgh structures
   struct mmsghdr mmsgh[elementCountP];
   struct iovec iov[elementCountP];
